angular/index.d.ts

```diff
@@ -756,7 +756,7 @@ declare namespace angular {
          * @param comparator Function used to determine the relative order of value pairs.
          * @return An array containing the items from the specified collection, ordered by a comparator function based on the values computed using the expression predicate.
          */
-        <T>(array: T[], expression: string|((value: T) => any)|(((value: T) => any)|string)[], reverse?: boolean, comparator?: IFilterOrderByComparatorFunc): T[];
```

Comment:
My motivation was pretty simple and un-nuanced. I am using filter in `.ts` code, and when I changed to start using `strictNullChecks` I got errors. In my case, all of the errors were because I was passing it things that could be `undefined` in addition to an array.